hack/generate.sh

```diff
@@ -110,22 +126,31 @@ run_target() {
         done
         
         if [[ ${#invalid_options[@]} -gt 0 ]]; then
-            printf "Skipping invalid options: %s, Available options are: %s\n\n" "${invalid_options[*]}" "${available_options[*]}"
+            printf "Skipping invalid options: %s, Available options are: %s\n\n" "${invalid_options[*]}" "${AVAILABLE_CODEGEN_OPTIONS[*]}"
         fi
       fi
 
       if [[ ${#valid_options[@]} -gt 0 ]]; then
-        printf "\n> Generating codegen for groups: ${valid_options[*]}\n\n"
+        printf "\n> Generating codegen for groups: %s\n" "${valid_options[*]}"
         $REPO_ROOT/hack/update-codegen.sh --which "${valid_options[*]}" --mode "$MODE"
+      else
+        printf "!! No valid groups provided for codegen, Available groups are: %s\n\n"  "${AVAILABLE_CODEGEN_OPTIONS[*]}"
       fi
       ;;
     manifests)
+      local which=$MANIFESTS_FOLDERS
+      if [[ -z "$which" ]]; then
+        which=("${DEFAULT_MANIFESTS_FOLDERS[@]}")
+      fi
+
+      printf "\n> Generating manifests for folders: %s\n" "${which[*]}"
       if [[ "$MODE" == "sequential" ]]; then
         # In sequential mode, paths need to be converted to go package notation (e.g., ./charts/...)
-        which=$(overwrite_paths)
-        $REPO_ROOT/hack/generate-sequential.sh $which
+        $REPO_ROOT/hack/generate-sequential.sh $(overwrite_paths "${which[@]}")
+      elif [[ "$MODE" == "parallel" ]]; then
+        $REPO_ROOT/hack/generate-parallel.sh "${which[@]}"
       else
-        $REPO_ROOT/hack/generate-parallel.sh $WHICH
+        printf "!! Invalid mode, Specify either 'parallel' or 'sequential'\n\n"
```

Comment:
```suggestion
  printf "ERROR: Invalid mode ('%s'). Specify either 'parallel' or 'sequential'\n\n" "$MODE"
```

We would suggest to `exit 1` in all the places we report an error and continue.